### PR TITLE
feat: [SSCA-838]: populate ssca service secrets on container step group initialization

### PR DESCRIPTION
**Pull request by inderschera**
## Describe your changes
- need to populate ssca service end point and ssca service secret in container step group initialization.
- Added a cache to avoid calls for same pipeline execution id again and again
- Not doing the changes during pod initialization in pipeline service - Closed the previous PR with the comment -  https://github.com/harness/harness-core/pull/55033#issuecomment-1795489135

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55057)

### CODE REVIEW

**Review by reetika04 — DISMISSED** [review]

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java [hunk 1]
```diff
@@ -75,6 +103,7 @@ public PluginCreationResponseWrapper getPluginInfo(
                                                .stream()
                                                .map(SecretNgVariableUtils::getSecretVariable)
                                                .collect(Collectors.toList());
+    secretVariables.addAll(getSscaServiceSecrets(ambiance));
```
Comment (dhruv-harness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:106]: will this be set for all ci plugins?

Comment (inderschera) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:106]: yes. Consider it as being set once for entire pod irrespective of the steps.
We don't want to put this logic during pod initialization in pipeline service else pipeline service would also need to be aware of SSCA. https://github.com/harness/harness-core/pull/55033#issuecomment-1795489135

**Comment by inderschera** [comment]
trigger uts

**Review by dhruv-harness — APPROVED** [review]

**Comment by dhruv-harness** [comment]
Lets try to add UTs

**Comment by inderschera** [comment]
trigger uts

File: 332-ci-manager/service/src/test/java/io/harness/ci/plugin/CiPluginStepInfoProviderTest.java [hunk 2]
```diff
@@ -222,4 +231,53 @@ public void testGetPluginInfoPluginCompatibleStep() {
...
+    PluginCreationRequest pluginCreationRequest =
+        PluginCreationRequest.newBuilder().setStepJsonNode(containerRunStepNode).setOsType("Linux").build();
+
+    PluginCreationResponseWrapper pluginInfo =
+        ciPluginStepInfoProvider.getPluginInfo(pluginCreationRequest, Set.of(1234), ambiance);
+    verify(sscaServiceUtils, times(1)).getSSCAServiceEnvVariables("accountId", "orgId", "projectId");
+    assertThat(pluginInfo.getResponse().getPluginDetails().getEnvVariablesWithPlainTextSecretMap())
+        .isEqualTo(envMapWithSecrets);
+
+    pluginInfo = ciPluginStepInfoProvider.getPluginInfo(pluginCreationRequest, Set.of(1234), ambiance);
```
Comment (dhruv-harness) [inline 332-ci-manager/service/src/test/java/io/harness/ci/plugin/CiPluginStepInfoProviderTest.java:278]: why are we asserting the same response twice

**Review by harshjain12 — APPROVED** [review]

File: 332-ci-manager/service/src/test/java/io/harness/ci/plugin/CiPluginStepInfoProviderTest.java (hunk 2 above)
Comment (inderschera) [reply 332-ci-manager/service/src/test/java/io/harness/ci/plugin/CiPluginStepInfoProviderTest.java:278]: Here, I am confirming if cache is working correctly. On getting plugin info twice, there was only one invocation to ssca manager.

**Pull request merged**